src/kv/test/kv_test.cpp

```diff
@@ -632,55 +630,56 @@ TEST_CASE("private recovery map swap")
   // We compact twice, deliberately. A public KV during recovery
   // would have compacted some number of times.
   {
-    Store::Tx tx;
+    ccf::Tx tx;
     auto v = tx.get_view(pub1);
     v->put(42, "42");
     tx.commit();
   }
   {
-    Store::Tx tx;
+    ccf::Tx tx;
     auto v = tx.get_view(pub1);
     v->put(42, "43");
     tx.commit();
   }
   s1.compact(s1.current_version());
   {
-    Store::Tx tx;
+    ccf::Tx tx;
     auto v = tx.get_view(pub1);
     v->put(44, "44");
     tx.commit();
   }
   s1.compact(s1.current_version());
   {
-    Store::Tx tx;
+    ccf::Tx tx;
     auto v = tx.get_view(pub1);
     v->put(45, "45");
     tx.commit();
   }
 
   INFO("Populate s2 with private entries");
   // We compact only once, at a lower index than we did for the public
-  // KV, which is what we expect during recovery of the private KV.
+  // KV, which is what we expect during recovery of the private KV. We do expect
+  // that the _entire_ private state is compacted
   {
-    Store::Tx tx;
+    ccf::Tx tx;
     auto v = tx.get_view(priv2);
     v->put(12, 12);
     tx.commit();
   }
-  s2.compact(s2.current_version());
   {
-    Store::Tx tx;
+    ccf::Tx tx;
     auto v = tx.get_view(priv2);
     v->put(13, 13);
     tx.commit();
   }
+  s2.compact(s2.current_version());
```

Comment:
This is a real change!

I've removed `set_read_committed` entirely since it was unused apart from in this unit test, and it should be replaceable with `get_globally_committed`. However, that broke this unit test. When we swap private maps, the `set_read_committed` method (creating views at only globally-committed read_versions) did what we wanted, but calling `get_globally_committed` returned a different value! Specifically, key 13 was _not_ present in s1's committed roll! There's a comment on `swap_private_maps` that describes exactly this behaviour, pasting verbatim:

```
     * It's also important to note that swapping in private state
     * may result in previously uncompacted writes becoming effectively
     * compacted, from a user's perspective (they would not be internally
     * compacted until the next compact() however). So it is important to
     * make sure that the private state being swapped in is fully compacted
     * before the swap.
```

So I've followed the guidance in the final sentence - the private state is _fully_ compacted before swapping. If anyone understands this better than I do and sees a problem, please shout!